subsys/bluetooth/host/hci_core.c

```diff
@@ -823,7 +838,12 @@ static void le_enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	 * Make lookup to check if there's a connection object in
 	 * CONNECT state associated with passed peer LE address.
 	 */
-	conn = bt_conn_lookup_state_le(&id_addr, BT_CONN_CONNECT);
+	for (u32_t i = 0; i < ARRAY_SIZE(le_states); i++) {
```

Comment:
Same here. Feel free to refactor into a small helper function if you think that's better.